lib/request.js

```diff
@@ -117,6 +117,8 @@ module.exports = {
   set path(path) {
     var url = parse(this.req);
     url.pathname = path;
+    url.path = url.pathname + (url.search || '');
```

Comment:
This is not the best idea; I had similar changes in my modules and the best solution is to do `url.path = null;` here.